clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesUnregisteredCustomResourceCachingAgent.java

```diff
@@ -50,4 +62,9 @@ public KubernetesUnregisteredCustomResourceCachingAgent(
   protected ImmutableList<KubernetesKind> primaryKinds() {
     return credentials.getCrds();
   }
+
+  @Override
+  protected boolean cachesKind(KubernetesKind kind) {
+    return !kind.getApiGroup().isNativeGroup();
```

Comment:
I guess I don't have the context in my head about what KubernetesUnregisteredCustomResourceCachingAgent does differently from KubernetesCoreCachingAgent...but I could use some comments here about why this method does what it does.

Reply:
Maybe you could do us all a big favor and add some comments to the class about what it's for, etc. ?

Reply:
Thank you!